tests/kafkatest/services/connect.py

```diff
@@ -107,45 +107,49 @@ def clean_node(self, node):
     def config_filenames(self):
         return [os.path.join(self.PERSISTENT_ROOT, "connect-connector-" + str(idx) + ".properties") for idx, template in enumerate(self.connector_config_templates or [])]
 
-    def list_connectors(self, node=None, retries=0, retry_backoff=.01):
-        return self._rest_with_retry('/connectors', node=node, retries=retries, retry_backoff=retry_backoff)
+    def list_connectors(self, node=None, **kwargs):
+        return self._rest_with_retry('/connectors', node=node, **kwargs)
 
-    def create_connector(self, config, node=None, retries=0, retry_backoff=.01):
+    def create_connector(self, config, node=None, **kwargs):
         create_request = {
             'name': config['name'],
             'config': config
         }
-        return self._rest_with_retry('/connectors', create_request, node=node, method="POST", retries=retries, retry_backoff=retry_backoff)
+        return self._rest_with_retry('/connectors', create_request, node=node, method="POST", **kwargs)
 
-    def get_connector(self, name, node=None, retries=0, retry_backoff=.01):
-        return self._rest_with_retry('/connectors/' + name, node=node, retries=retries, retry_backoff=retry_backoff)
+    def get_connector(self, name, node=None, **kwargs):
+        return self._rest_with_retry('/connectors/' + name, node=node, **kwargs)
 
-    def get_connector_config(self, name, node=None, retries=0, retry_backoff=.01):
-        return self._rest_with_retry('/connectors/' + name + '/config', node=node, retries=retries, retry_backoff=retry_backoff)
+    def get_connector_config(self, name, node=None, **kwargs):
+        return self._rest_with_retry('/connectors/' + name + '/config', node=node, **kwargs)
 
-    def set_connector_config(self, name, config, node=None, retries=0, retry_backoff=.01):
-        return self._rest_with_retry('/connectors/' + name + '/config', config, node=node, method="PUT", retries=retries, retry_backoff=retry_backoff)
+    def set_connector_config(self, name, config, node=None, **kwargs):
+        # Unlike many other calls, a 409 when setting a connector config is expected if the connector already exists.
+        # However, we also might see 409s for other reasons (e.g. rebalancing). So we still perform retries at the cost
```

Comment:
@hachikuji Yeah, we have an extra code in the JSON, but it's just a 409 in both cases. In things like REST proxy, we use the extended error code to indicate specific ways the error can occur so things don't get too messy. HTTP status codes are, unfortunately, a bit restrictive, so its hard not to end up with this kind of conflict when you have both general errors that can happen across a lot of API endpoints and specific ones for a single endpoint which happen to need the same error code.